test/API-functional/device-spec.js

```diff
@@ -0,0 +1,126 @@
+/* eslint no-undef: "off" */
+const assert = require('assert').strict
+const StApi = require('../../lib/api')
+// User token for prod env
+const opts = {authToken: '0594f1d2-69e9-4809-a773-5fe7268e3f01'}
+const api = new StApi(opts)
+
+let params
+let devicesApi
+let deviceId
+accessToken = '8ede1b8e-890e-4995-9ee8-ea4d06d885f5'
```

Comment:
How's this different than `opts.authToken`, defined above?

Reply:
In here, authToken is a user token. However for installing a device API (probably execute commands and events too) requires smartapp token (as it's a smartapp managed device) which has expiry of 5min. So needed to generate a refresh token (using client ID, client secret, and existing refresh token) to generate a new one and execute api calls with this new token. Also, this is where we can embed this as an env variable in circle ci .

Reply:
I'm a little concerned about requiring (but hiding) real production tokens in this project, as anybody who wants to contribute will not be able to run or pass tests. It will discourage them from including tests entirely. We can put them into CI, but then they can only pass those tests in CI, so I'm thinking that our `test` script needs to only attempt these _functional_ tests in a CI environment. 

In the event that you need to put them into the environment: 
1. This project does not have a `dotenv` support
2. You can add the lib as a `devDependency`
3. Add an `.env.example`
4. Configure it to obtain the token from `process.env.YOUR_AUTH_TOKEN` instead